sdk/core/core/inc/az_http_transport.h

```diff
@@ -85,27 +85,97 @@ AZ_NODISCARD az_result
  *
- * This function is expected to be used by transport layer only.
+ * @remarks This function is expected to be used by transport layer only.
  *
- * @param request HTTP request to get parts from.
- * @param[out] out_method __[nullable]__ Pointer to write HTTP method to. Use `NULL` to ignore
- * getting this value.
- * @param[out] out_url __[nullable]__ Pointer to write URL to. Use `NULL` to ignore getting this
- * value.
- * @param[out] out_body __[nullable]__ Pointer to write HTTP request body to. Use `NULL` to ignore
- * getting this value.
+ * @param[in] request HTTP request to get parts from.
+ * @param[out] out_method Pointer to write HTTP method to.
  *
- * @retval AZ_OK Success.
+ * @retval An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
+ */
+AZ_NODISCARD az_result
+az_http_request_get_method(_az_http_request const* request, az_http_method* out_method);
+
+/**
+ * @brief Get url from an HTTP request.
+ *
+ * @remarks This function is expected to be used by transport layer only.
+ *
+ * @param[in] request HTTP request to get parts from.
+ * @param[out] out_url Pointer to write URL to.
+ *
+ * @retval An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
+ */
+AZ_NODISCARD az_result az_http_request_get_url(_az_http_request const* request, az_span* out_url);
+
+/**
+ * @brief Get body from an HTTP request.
+ *
+ * @remarks This function is expected to be used by transport layer only.
+ *
+ * @param[in] request HTTP request to get parts from.
+ * @param[out] out_body Pointer to write HTTP request body to.
+ *
+ * @retval An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
+ */
+AZ_NODISCARD az_result az_http_request_get_body(_az_http_request const* request, az_span* out_body);
+
+/**
+ * @brief This function is expected to be used by transport adapters like curl. Use it to write
+ * content from \p source to \p response.
+ *
+ * @remarks Parameter \p source can be an empty span. If so, nothing will be written.
+ *
+ * @param[in] response Pointer to an az_http_response.
+ * @param[in] source This is an az_span with the content to be written into response.
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if successful
+ *         - #AZ_ERROR_INSUFFICIENT_SPAN_SIZE if the \p response is not big enough to contain the
+ * \p source content
  */
-AZ_NODISCARD az_result az_http_request_get_parts(
-    _az_http_request const* request,
-    az_http_method* out_method,
-    az_span* out_url,
-    az_span* out_body);
+AZ_NODISCARD az_result az_http_response_write_span(az_http_response* response, az_span source);
+
+/**
+ * @brief This function is expected to be used by transport adapters like curl. Use it to source
+ * just one single byte to an az_http_response.
+ *
+ * @remarks This is a convenient way of hiding the internal implementation of az_htt_response.
```

Comment:
Missed the previous feedback to fix typo (from https://github.com/Azure/azure-sdk-for-c/pull/629#discussion_r416209631):
```suggestion
 * @remarks This is a convenient way of hiding the internal implementation of az_http_response.
```